File: backend/bondi_location_service.py
import requests
import datetime
import os
import json
from dotenv import load_dotenv

load_dotenv()

CLIENT_ID = os.getenv("CLIENT_ID")
CLIENT_SECRET = os.getenv("CLIENT_SECRET")

ROUTES = [283, 284, 287, 288, 496, 497, 922, 923, 924, 925, 926, 927, 1161, 1162, 1199, 1200, 1201]
AGENCIES = [87, 83, 51]

def obtener_ubicaciones():
    ubicaciones = []

    # Obtener y guardar la respuesta en el archivo
    url = (
        f"https://apitransporte.buenosaires.gob.ar/colectivos/vehiclePositionsSimple"
        f"?client_id={CLIENT_ID}&client_secret={CLIENT_SECRET}"
    )

    try:
        res = requests.get(url)
        with open("response.json", "w", encoding="utf-8") as archivo:
            json.dump(res.json(), archivo, ensure_ascii=False, indent=4)
    except Exception as e:
        print(f"[Error] al guardar la respuesta: {e}")
        return []

    # Leer desde el archivo response.json
    try:
        with open("response.json", "r", encoding="utf-8") as archivo:
            data = json.load(archivo)

        if isinstance(data, list) and len(data) > 0:
            for item in data:
                # if (item.get("route_id") in ROUTES) or (item.get("agency_id") in AGENCIES):
                    ubicaciones.append({
                        "route_short_name": item["route_short_name"],
                        "agency_id": item["agency_id"],
                        "latitude": item["latitude"],
                        "longitude": item["longitude"],
                        "speed": item["speed"],
                        "timestamp": datetime.datetime.fromtimestamp(item["timestamp"]).strftime("%H:%M:%S"),
                        "trip_headsign": item["trip_headsign"]
                    })
        else:
            print("[Aviso] No se encontraron vehículos.")
    except Exception as e:
        print(f"[Error] al leer el archivo JSON: {e}")

    return ubicaciones

File: backend/bondi_location_service.js
const axios = require("axios");
const dotenv = require("dotenv");
dotenv.config();

const CLIENT_ID = process.env.CLIENT_ID;
const CLIENT_SECRET = process.env.CLIENT_SECRET;

const ROUTES = [283, 284, 287, 288, 496, 497, 922, 923, 924, 925, 926, 927, 1161, 1162, 1199, 1200, 1201];
const AGENCIES = [87, 83, 51];

async function obtenerUbicaciones() {
  const ubicaciones = [];

  const url = `https://apitransporte.buenosaires.gob.ar/colectivos/vehiclePositionsSimple?client_id=${CLIENT_ID}&client_secret=${CLIENT_SECRET}`;

  try {
    const res = await axios.get(url);

    if (res.status === 200) {
      const data = res.data;

      if (Array.isArray(data) && data.length > 0) {
        data.forEach((item) => {

          // if (ROUTES.includes(Number(item.route_id)) || AGENCIES.includes(Number(item.agency_id))) {
          ubicaciones.push({
            route_short_name: item.route_short_name,
            agency_id: item.agency_id,
            latitude: item.latitude,
            longitude: item.longitude,
            timestamp: new Date(item.timestamp * 1000).toLocaleTimeString("en-US", { hour12: false }),
            trip_headsign: item.trip_headsign,
          });
          // }
        });
      } else {
        console.log("[Aviso] No se encontraron vehículos.");
      }
    } else {
      console.log(`[Error] La API respondió con un error. Código: ${res.status}`);
    }
  } catch (error) {
    console.log(`[Error] Error al hacer la solicitud a la API: ${error}`);
  }

  return ubicaciones;
}

module.exports = { obtenerUbicaciones };
